x/stakeibc/keeper/deposit_records.go

```diff
@@ -66,23 +68,19 @@ func (k Keeper) TransferExistingDepositsToHostZones(ctx sdk.Context, epochNumber
 		}
 		delegateAddress := delegateAccount.GetAddress()
 
-		blockHeight, found := k.GetLightClientHeightSafely(ctx, hostZone.ConnectionId)
-		if !found {
-			k.Logger(ctx).Error(fmt.Sprintf("Could not find blockHeight for host zone %s, aborting transfers to host zone this epoch", hostZone.ConnectionId))
-			continue
-		} else {
-			k.Logger(ctx).Info(fmt.Sprintf("Found blockHeight for host zone %s: %d", hostZone.ConnectionId, blockHeight))
-		}
-
-		timeoutHeight := clienttypes.NewHeight(0, blockHeight+ibcTimeoutBlocks)
 		transferCoin := sdk.NewCoin(hostZone.GetIBCDenom(), sdk.NewInt(depositRecord.Amount))
-		// QUESTION: Is there a good place to store "tranfer" as a constant?
-		msg := ibctypes.NewMsgTransfer("transfer", hostZone.TransferChannelId, transferCoin, hostZoneModuleAddress, delegateAddress, timeoutHeight, 0)
+		// timeout 30 min in the future
+		// NOTE: this assumes no clock drift between chains, which tendermint guarantees
+		// if we onboard non-tendermint chains, we need to use the time on the host chain to
+		// calculate the timeout
+		// https://github.com/tendermint/tendermint/blob/v0.34.x/spec/consensus/bft-time.md
+		timeoutTimestamp := ctx.BlockTime().UnixNano() + int64(feeTransferTimeoutNanos)
+		msg := ibctypes.NewMsgTransfer(ibctransfertypes.PortID, hostZone.TransferChannelId, transferCoin, hostZoneModuleAddress, delegateAddress, clienttypes.Height{}, uint64(timeoutTimestamp))
```

Comment:
_extremely_ nit, so I apologize in advance, but might be better to cast `ctx.BlockTime().UnixNano()` in line 77 to uint64 rather than casting on line 78 (1 cast vs 2 casts)

Reply:
great call! done